pkg/controller/node/cidr_allocator.go

```diff
@@ -139,11 +162,19 @@ func (r *rangeAllocator) occupyCIDR(node *api.Node) error {
 // AllocateOrOccupyCIDR looks at the given node, assigns it a valid CIDR
 // if it doesn't currently have one or mark the CIDR as used if the node already have one.
 func (r *rangeAllocator) AllocateOrOccupyCIDR(node *api.Node) error {
+	if node == nil {
+		return nil
+	}
+	if !r.leaseNode(node.Name) {
+		glog.V(2).Infof("Node %v is already in a process of CIDR assignment.", node.Name)
+		return nil
+	}
 	if node.Spec.PodCIDR != "" {
 		return r.occupyCIDR(node)
 	}
 	podCIDR, err := r.cidrs.allocateNext()
```

Comment:
Quick question : when `r.cidrs.allocateNext()` returns an error, we don't call `r.releaseNode(node.Name)`. Is this by design?